tests/zfs-tests/tests/functional/cli_root/zfs_unshare/zfs_unshare_007_pos.ksh

```diff
@@ -0,0 +1,86 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright 2016, loli10K. All rights reserved.
+# Copyright (c) 2018 by Delphix. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+
+#
+# DESCRIPTION:
+# Verify that 'zfs destroy' on a shared dataset, will unshare it.
+#
+# STRATEGY:
+# 1. Create and share datasets with different protocols.
+# 2. Verify these datasets are shared.
+# 3. Invoke 'zfs destroy' on these datasets.
+# 4. Verify these datasets are not shared.
+#
+
+verify_runnable "global"
+
+if is_linux; then
+	log_unsupported "some distros come with Samba "user shares" disabled"
```

Comment:
nit: Oddly the quoting around "user shares" doesn't seem to cause a problem in the logs, but now would be a good opportunity to switch it to say single quotes.  Here's and in `zfs_unshare_006_pos`.

Reply:
I'd rather no skip it on Linux. I did this since I saw it used elsewhere (i.e. `zfs_unshare_006_pos`) with the hopes that it'd somehow enable it on test systems that had Samba installed.

If this will never get run as a result, then I'd rather remove this check, along with the Samba specific parts of this test. There's no point in adding a test if it'll always be disabled.

Reply:
That would be my preference as well, tests which don't run regularly aren't of much use.  @loli10K what do you think of removing this functionality from `zfs_unshare_006_pos` as well so it can be enabled by default.